### ISSUE TITLE
Inline message stops receiving taps after its host Fragment view is recreated (EngineWebView never re-subscribes to the new lifecycle owner)

### ISSUE THREAD

**Issue by ShanavasPS** [customerio/customerio-android#874]
**Environment:** messaging-in-app 4.18.2 (the same code is on `main`), via customerio-reactnative 6.5.2, React Native 0.86 (Fabric), react-native-screens 4.25.2 native stack, androidx.fragment 1.8.9, Android 16 (Pixel 9a).

**Steps**
1. Host `InlineInAppMessageView` (through the React Native `InlineInAppMessageView`) on a screen of a `@react-navigation/native-stack` navigator and let an inline message embed.
2. Push any other screen on top of it.
3. Go back.
4. Tap the message's action button or its close button.

**Expected:** the tap is handled (action dispatched / message dismissed).

**Actual:** nothing happens. No `Received event from WebView` line, no `tap`, no click metric. The message is still rendered, its JavaScript handlers are intact and `message.performAction(...)` returns without error. A message embedded *after* the navigation works; the damage is per `EngineWebView` instance and lasts until re-embed or process restart. Backgrounding/foregrounding the app does **not** cause it.

**Cause**

`EngineWebView.setup()` calls `viewLifecycleOwner?.addObserver(this)` exactly once, on whatever `ViewTreeLifecycleOwner` it finds — under a native stack that is the *view* lifecycle owner of the react-native-screens `ScreenFragment` hosting the screen, not the Activity.

When a screen is pushed, react-native-screens removes the Fragment below the top (`ScreenStack.onUpdate`). That Fragment's view lifecycle goes `ON_PAUSE` → `EngineWebView.onLifecyclePaused()` → `EngineWebViewInterface.detach()` sets `isAttachedToWebView = false` and removes the JavaScript interface, and is then destroyed. On pop the Fragment gets a **new** view lifecycle owner, but the same `EngineWebView` instance survives (React Native owns the view and react-native-screens re-parents it), and it is still registered on the destroyed owner. `ON_RESUME` never arrives, `isAttachedToWebView` stays `false`, and `EngineWebViewInterface.postMessage` drops every event at its first line:

```kotlin
if (!isAttachedToWebView) return
```

`BaseInlineInAppMessageView` already re-registers *its* observer in `onAttachedToWindow` (#611 / MBL-1329), which is why the store side survives return navigation — `EngineWebView` was never given the same treatment.

Log signature of a broken round trip (SDK at debug level): a single `EngineWebView onLifecyclePaused` at the push and no `EngineWebView onLifecycleResumed` after the pop.

**Suggested fix**

In `EngineWebView`, override `onAttachedToWindow` to `viewLifecycleOwner?.addObserver(this)` (and `onDetachedFromWindow` to remove it), or re-attach the interface on window attach. `LifecycleRegistry.addObserver` replays the owner's current state, so a RESUMED owner immediately re-attaches the interface.

**Workaround we verified** (in the React Native wrapper's `ReactInlineInAppMessageView`, applied as a patch):

```kotlin
override fun onAttachedToWindow() {
    super.onAttachedToWindow()
    val lifecycle = findViewTreeLifecycleOwner()?.lifecycle ?: return
    for (i in 0 until childCount) (getChildAt(i) as? LifecycleObserver)?.let(lifecycle::addObserver)
}

override fun onDetachedFromWindow() {
    findViewTreeLifecycleOwner()?.lifecycle?.let { l ->
        for (i in 0 until childCount) (getChildAt(i) as? LifecycleObserver)?.let(l::removeObserver)
    }
    super.onDetachedFromWindow()
}
```

With it, the pop logs `EngineWebView onLifecycleResumed` and taps reach native again on the same device and build.

**Comment by ShanavasPS** [comment]
Proposed fix in #875: once `setup()` has run, `EngineWebView` re-subscribes to the current view-tree lifecycle owner in `onAttachedToWindow` (and unsubscribes in `onDetachedFromWindow`), plus a Robolectric test that replaces the owner around a detach/re-attach and fails on the current code. The equivalent re-subscribe applied from the React Native wrapper's host view is verified on device (Pixel 9a, Android 16): the push/pop round trip now logs `EngineWebView onLifecycleResumed` and taps reach `tap()` again.
